Assets/Scripts/Minigames/Mastermind/Mastermind.cs

```diff
@@ -1,88 +1,334 @@
+            for (int i = 0; i < rowItemsToVerify.Length; i++)
             {
-                if (code[i] == SecretCodeTemp[i])
-                {
-                    good++;
-                    code[i] = "good";
-                    SecretCodeTemp[i] = "good";
+                rowItemsToVerify[i] = rowToVerify.transform.Find("c" + (i + 1)).gameObject;
+            }
 
-                }
+            // Get the verify icons. These are the circles that are in the row and have the "Verif" tag
+            // These are the circles that will be used to show the player how many correct colors and positions they have
+            // Brackets are used to scope index variable to the loop
+            for (int i = 0; i < rowVerifyIcons.Length; i++)
+            {
+                rowVerifyIcons[i] = rowToVerify.transform.Find("v" + (i + 1)).gameObject;
+            }
+
+            // check if the row is filled. If any of the circles are empty, return
+            if (rowItemsToVerify.Any(rowItem => rowItem.GetComponent<Image>().sprite == Empty)) return;
+
+            // Set the verify icons to the correct amount of green and yellow circles. Heart = correct position,
+            // yellow = correct shape. Fill the rest with red circles
+            int goodPositions = GetAmountOfCorrectPositions(_playerInputCode, _secretCodeToGuess);
+            int goodColors = GetAmountOfCorrectColors(_playerInputCode, _secretCodeToGuess);
+
+            for (int i = 0; i < goodPositions; i++)
+            {
+                rowVerifyIcons[i].GetComponent<Image>().sprite = CorrectPosition;
+            }
+
+            for (int i = goodPositions; i < goodColors + goodPositions; i++)
+            {
+                rowVerifyIcons[i].GetComponent<Image>().sprite = CorrectColor;
+            }
+
+            for (int i = goodColors + goodPositions; i < Cols; i++)
+            {
+                rowVerifyIcons[i].GetComponent<Image>().sprite = Incorrect;
+            }
+
+            if (goodPositions == 4)
+            {
+                CompleteGameStep();
+                return;
             }
-            Array.Copy(code, _codePlayer, code.Length);
-            return good;
+
+            if (_currentRow == 9)
+            {
+                Lose();
+                return;
+            }
+
+            // TODO: fix shape for selected row
+            _currentRow++;
+
+            Color originMastermindColor = rowToVerify.GetComponent<Image>().color;
+            Color selectionMastermindColor = originMastermindColor;
+            selectionMastermindColor.a = 0.25f;
+
+            rowToVerify.GetComponent<Image>().color = selectionMastermindColor;
+            previousRow.GetComponent<Image>().color = originMastermindColor;
         }
 
+        void Win()
+        {
+            HiddenCodeRow.SetActive(false);
+        }
+
+        void Lose()
+        {
+            HiddenCodeRow.SetActive(false);
+        }
```

Comment:
word dit nog uitgebreid of is dit duplicate code?

Reply:
Methods have been made unique